worker/internal/infrastructure/aws/s3.go

```diff
@@ -0,0 +1,148 @@
+package aws
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"path"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/reearth/reearth-cms/worker/internal/usecase/gateway"
+	"github.com/reearth/reearthx/log"
+	"github.com/reearth/reearthx/rerror"
+)
+
+const (
+	s3AssetBasePath string = "assets"
+)
+
+type fileRepo struct {
+	bucketName   string
+	cacheControl string
+	s3Client     *s3.Client
+	s3Uploader   *manager.Uploader
+}
+
+func NewFile(ctx context.Context, bucketName, cacheControl string) (gateway.File, error) {
+	if bucketName == "" {
+		return nil, errors.New("bucket name is empty")
+	}
+
+	cfg, err := config.LoadDefaultConfig(ctx)
+	if err != nil {
+		log.Errorf("aws: failed to create config: %v\n", err)
+		return nil, rerror.ErrInternalBy(err)
+	}
+
+	s3Client := s3.NewFromConfig(cfg)
+	s3Uploader := manager.NewUploader(s3Client)
+
+	return &fileRepo{
+		bucketName:   bucketName,
+		cacheControl: cacheControl,
+		s3Client:     s3Client,
+		s3Uploader:   s3Uploader,
+	}, nil
+}
+
+func (f *fileRepo) Read(ctx context.Context, filePath string) (gateway.ReadAtCloser, int64, int64, error) {
+	if filePath == "" {
+		return nil, 0, 0, rerror.ErrNotFound
+	}
+
+	objectKey := getS3ObjectKeyFromURL(s3AssetBasePath, filePath)
+
+	params := &s3.GetObjectInput{
+		Bucket: &f.bucketName,
+		Key:    &objectKey,
+	}
+
+	resp, err := f.s3Client.GetObject(ctx, params)
+	if err != nil {
+		log.Errorf("aws: read object err: %+v\n", err)
+		return nil, 0, 0, rerror.ErrInternalBy(err)
+	}
+
+	proceeded, _ := strconv.ParseInt(resp.Metadata["proceeded"], 10, 64)
+
+	// Read all data into memory
+	objectData, err := io.ReadAll(resp.Body)
```

Comment:
As I pointed out before, if you load a large file, the server will go down due to lack of memory in one shot.

It is a good idea to analyze the cause of the need for io.ReaderAt and improve it so that ReaderAt is not needed. If you are still not sure, please consult with us again.

Reply:
https://github.com/markandrus/s3readerat may be helpful

Reply:
It appears that the `readerAt` format is required as an input for the `archive/zip` package in order to read from a specific offset. 
Currently, our implementation on GCP follows a similar approach by reading everything into memory.

Could you please provide suggestions on how to fix the Read method for both AWS and GCP?

Reply:
Hmm current implementation is OK

We can implement a way to prevent loading the entire file at a later date. The simplest way is to write to a file. Then we can consider using:

- https://github.com/markandrus/s3readerat
- https://github.com/snabb/httpreaderat
- https://github.com/avvmoto/buf-readerat

Reply:
Well, that makes sense. I think that `buf-readerat` or a similar package could come in handy to solve this problem in the future.

Thanks!